src/main/java/com/sinthoras/visualprospecting/mixins/xaerosworldmap/GuiMapMixin.java

```diff
@@ -0,0 +1,194 @@
+package com.sinthoras.visualprospecting.mixins.xaerosworldmap;
+
+import com.sinthoras.visualprospecting.VP;
+import com.sinthoras.visualprospecting.gui.model.MapState;
+import com.sinthoras.visualprospecting.gui.model.layers.LayerManager;
+import com.sinthoras.visualprospecting.gui.xaeromap.XaeroMapState;
+import com.sinthoras.visualprospecting.gui.xaeromap.buttons.LayerButton;
+import com.sinthoras.visualprospecting.gui.xaeromap.buttons.SizedGuiTexturedButton;
+import com.sinthoras.visualprospecting.gui.xaeromap.renderers.InteractableLayerRenderer;
+import com.sinthoras.visualprospecting.gui.xaeromap.renderers.LayerRenderer;
+import com.sinthoras.visualprospecting.gui.xaeromap.rendersteps.RenderStep;
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.gui.GuiButton;
+import net.minecraft.client.gui.GuiScreen;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.util.ResourceLocation;
+import org.objectweb.asm.Opcodes;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Shadow;
+import org.spongepowered.asm.mixin.Unique;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.Slice;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable;
+import org.spongepowered.asm.mixin.injection.callback.LocalCapture;
+import xaero.map.MapProcessor;
+import xaero.map.gui.CursorBox;
+import xaero.map.gui.GuiMap;
+import xaero.map.gui.ScreenBase;
+import xaero.map.misc.Misc;
+
+// stop yelling at me it just works
+@SuppressWarnings({"MixinAnnotationTarget", "UnresolvedMixinReference"})
+@Mixin(value = GuiMap.class, remap = false)
+public abstract class GuiMapMixin extends ScreenBase {
+
+	@Unique private int oldMouseX = 0;
+    @Unique private int oldMouseY = 0;
+	@Unique private long timeLastClick = 0;
+
+	protected GuiMapMixin(GuiScreen parent, GuiScreen escape) {
+		super(parent, escape);
+	}
+
+	@Shadow private double cameraX;
+
+	@Shadow private double cameraZ;
+
+	@Shadow private double scale;
+
+	@Shadow public abstract void addGuiButton(GuiButton b);
```

Comment:
abstract instead of throwing a runtime exception. Nice touch. Not sure I prefer it though. Won't ever make a difference

Reply:
That's just how the mcdev plugin auto-adds shadow methods when you autocomplete them